ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -56,7 +77,201 @@
 
   @Override
   protected int command() {
-    throw new UnsupportedOperationException();
+    if (!validateConfigFilePresent()) {
+      return 1;
+    }
+
+    final MigrationConfig config;
+    try {
+      config = MigrationConfig.load(configFile);
+    } catch (KsqlException | MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    return command(
+        config,
+        MigrationsUtil::getKsqlClient,
+        getMigrationsDirFromConfigFile(configFile),
+        Clock.systemDefaultZone()
+    );
+  }
+
+  @VisibleForTesting
+  int command(
+      final MigrationConfig config,
+      final Function<MigrationConfig, Client> clientSupplier,
+      final String migrationsDir,
+      final Clock clock
+  ) {
+    final Client ksqlClient;
+    try {
+      ksqlClient = clientSupplier.apply(config);
+    } catch (MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    if (ValidateMigrationsCommand.validate(config, migrationsDir, ksqlClient)
+        && apply(config, ksqlClient, migrationsDir, clock)) {
+      ksqlClient.close();
+      return 0;
+    } else {
+      ksqlClient.close();
+      return 1;
+    }
+  }
+
+  private boolean apply(
+      final MigrationConfig config,
+      final Client ksqlClient,
+      final String migrationsDir,
+      final Clock clock
+  ) {
+    String previous = MetadataUtil.getLatestMigratedVersion(config, ksqlClient);
+    final int minimumVersion = previous.equals(MetadataUtil.NONE_VERSION)
+        ? 1
+        : Integer.parseInt(previous) + 1;
+    LOGGER.info("Loading migration files");
+    final List<Migration> migrations = getAllMigrations(migrationsDir).stream()
+        .filter(migration -> {
+          if (version > 0) {
+            return migration.getVersion() <= version && migration.getVersion() >= minimumVersion;
+          } else {
+            return migration.getVersion() >= minimumVersion;
+          }
+        })
+        .collect(Collectors.toList());
+
+    if (migrations.size() == 0) {
+      LOGGER.error("No eligible migrations found.");
```

Comment:
I don't think we should return an error if there are no migrations and the mode is `until` or `all`. We should check that we have log messages that make it clear which file(s) are being applied, and simply print a message saying there's nothing to apply in these modes, but I think the exit code should still be 0 rather than 1.